### PR TITLE
Convert SSL.SSLError to ssl.SSLError and close connection on error

### PR DESCRIPTION
**Pull request by chaen**

### CODE REVIEW

**Comment by fstagni** [comment]
Is this in preparation for python 3.10?

**Comment by chaen** [comment]
no no, it's for more stability for us, since we start using it more

File: tornado_m2crypto/m2iostream.py [hunk 1]
```diff
@@ -172,31 +172,35 @@ def _do_ssl_handshake(self):
...
+                # self._handshake_writing = True
                 return
             if res < 0:
                 err_num = self.socket.ssl_get_error(res)
                 gen_log.error("Err: %s" % err_num)
                 gen_log.error("Err Str: %s" % Err.get_error_reason(err_num))
                 return self.close()
-        except SSL.SSLError as e:
-            raise
+        except SSL.SSLError as err:
```
Comment (fstagni) [inline tornado_m2crypto/m2iostream.py:190]: Shouldn't you have done it also here?

Comment (chaen) [reply tornado_m2crypto/m2iostream.py:190]: No, because here I do not propagate it, I handle it

**Pull request merged**